### PR TITLE
Fix Admin portal looping in Safari issue

### PR DESCRIPTION
**Pull request by brionmario**
## Purpose
The purpose of this PR is to fix #https://github.com/LSFLK/MedicinesforLK/issues/363

## Goals

The admin portal was going in to infinite loop in Safari browsers.
When analyzed, it was figured out that the application makes two token requests which causes a race condition.

Screenshot 2022-08-22 at 11 55 02

In the `config`, silent sign in is enabled[1] which sends a sign in request silently during the application loading time.
Also, in the `app.tsx`, there is another sign in request[2].

[1] https://github.com/LSFLK/MedicinesforLK/blob/fba0e82fe95a1dd16b24a4675b89171a1a542c8b/admin-portal/src/config.ts#L17
[2] https://github.com/LSFLK/MedicinesforLK/blob/fba0e82fe95a1dd16b24a4675b89171a1a542c8b/admin-portal/src/App.tsx#L30

## Approach

1. Removed enabling automatic silent sign from the [config](https://github.com/LSFLK/MedicinesforLK/blob/fba0e82fe95a1dd16b24a4675b89171a1a542c8b/admin-portal/src/config.ts#L17).

2. Added the `trySignInSilently` back to the `app.tsx`.

3. Bump `@asgardeo/auth-react` SDK, to `v1.1.16`.
    - Fix looping issue due to React 18, Strict Mode.
        - https://github.com/asgardeo/asgardeo-auth-react-sdk/issues/125

4. Additionally, brought back `web worker`` storage which was disabled due to https://github.com/asgardeo/asgardeo-auth-react-sdk/issues/121.

### Screenshots
Aug-22-2022 16-31-12

## Related PRs
- N/A

## Learning
- N/A

### CODE REVIEW

File: admin-portal/package.json [hunk 1]
```diff
@@ -3,9 +3,9 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@asgardeo/auth-react": "^1.1.11",
+    "@asgardeo/auth-react": "^1.1.16",
     "@asgardeo/auth-spa": "^0.4.4",
-    "@asgardeo/token-exchange-plugin": "^0.1.3",
+    "@asgardeo/token-exchange-plugin": "0.1.3",
```
Comment (brionmario) [inline admin-portal/package.json:8]: Had to stick the version due to the [breaking changes](https://github.com/asgardeo/asgardeo-auth-spa-token-exchange/commit/0840597ace6964c42434873c36ec7bb42d33544f) in `0.1.4`.

@yathindrakodithuwakku please advice the team on the migration.

File: admin-portal/src/App.tsx [hunk 2]
```diff
@@ -41,15 +34,32 @@ function App() {
     MedicalNeedsService.http = http;
     PledgeService.http = http;
     SupplierService.http = http;
-  }, [state.isAuthenticated]);
+  }, []);
```
Comment (brionmario) [inline admin-portal/src/App.tsx:37]: Moved the `http` client instantiation to a standalone hook. IMO, we don't need to reinitialize when the user's authenticated state changes.

Please validate.

File: admin-portal/src/config.ts [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+import { AuthReactConfig, Storage } from "@asgardeo/auth-react";
+import { STSClientConfig } from "@asgardeo/token-exchange-plugin";
+
+const SDKConfig: AuthReactConfig & STSClientConfig = {
   clientID: "W_ZaTLGR20OKy0HTYPqSAJvuS5Ia",
   baseUrl: "https://api.asgardeo.io/t/elixir",
   signInRedirectURL: `${process.env.REACT_APP_SIGN_IN_REDIRECT_URL}`,
   signOutRedirectURL: `${process.env.REACT_APP_SIGN_OUT_REDIRECT_URL}`,
   scope: ["openid", "email", "profile"],
+  storage: Storage.WebWorker,
```
Comment (yathindrakodithuwakku) [inline admin-portal/src/config.ts:10]: In order to use webworker as the storage mechanism, this should be addressed beforehand IMO https://github.com/asgardeo/asgardeo-auth-react-sdk/issues/120

Comment (brionmario) [reply admin-portal/src/config.ts:10]: Let's revert this until then.

Comment (brionmario) [reply admin-portal/src/config.ts]: Fixed with https://github.com/LSFLK/MedicinesforLK/pull/430/commits/4759273784f0223c547229f90457537142cdfa32.

**Review by anjula-sack — APPROVED** [review]
Thank you! @brionmario

**Comment by brionmario** [comment]
> I am facing similar issue of reloading but it's coming due to call `/oauth2/jwks` giving `Referrer Policy: strict-origin-when-cross-origin`.
> 
> `Access to fetch at 'https://ids.test.dubaipolice.com/oauth2/jwks' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`
> 
> image

What's the `baseUrl` configured in the `AuthProvider` `config` object?

**Comment by liferayst** [comment]
> > I am facing similar issue of reloading but it's coming due to call `/oauth2/jwks` giving `Referrer Policy: strict-origin-when-cross-origin`.
> > `Access to fetch at 'https://ids.test.dubaipolice.com/oauth2/jwks' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`
> > image
> 
> What's the `baseUrl` configured in the `AuthProvider` `config` object?

"baseUrl": "https://ids.test.dubaipolice.com",

**Comment by brionmario** [comment]
@liferayst 

Also, please check if `http(s)://localhost:300` allowed in the origins section under the `protocol` tab of the Application in the Console.

Screenshot 2023-04-04 at 13 21 38

**Comment by brionmario** [comment]
> > @liferayst
> > Also, please check if `http(s)://localhost:3000` allowed in the `Allowed origins` section under the `protocol` tab of the Application in the Console.
> > Screenshot 2023-04-04 at 13 21 38
> 
> I am not able to find this option? Can you guide me in detail

May I know the WSO2 Identity Server version you're currently using?

**Comment by liferayst** [comment]
> > > @liferayst
> > > Also, please check if `http(s)://localhost:3000` allowed in the `Allowed origins` section under the `protocol` tab of the Application in the Console.
> > > Screenshot 2023-04-04 at 13 21 38
> > 
> > 
> > I am not able to find this option? Can you guide me in detail
> 
> May I know the WSO2 Identity Server version you're currently using?
6.0 we are using

**Comment by brionmario** [comment]
> > > > @liferayst
> > > > Also, please check if `http(s)://localhost:3000` allowed in the `Allowed origins` section under the `protocol` tab of the Application in the Console.
> > > > Screenshot 2023-04-04 at 13 21 38
> > > 
> > > 
> > > I am not able to find this option? Can you guide me in detail
> > 
> > 
> > May I know the WSO2 Identity Server version you're currently using?
> > 6.0 we are using

To allow CORS, you can use any of the following methods based on your preference.

1. Use the BETA Console (https://is.docs.wso2.com/en/6.0.0/references/about-this-release/#beta-features)
2. Allow the URL via the `deployment.toml` (https://is.docs.wso2.com/en/6.0.0/deploy/configure-cors/#configuring-cors-during-deployment)
3. Use the CORS REST API (https://is.docs.wso2.com/en/6.0.0/apis/cors-rest-api/)

IMO, to quickly test if this works, you can go with the `2` option since you have a test environment.

**Pull request merged**